test/client/fixtures/sandbox/shadow-ui-test.js

```diff
@@ -835,3 +835,34 @@ test('getElementsByTagName', function () {
     strictEqual(elements[0], textarea1);
     strictEqual(elements[1], void 0);
 });
+
+test('live collection should properly change if element was added (gh-1111)', function () {
+    var testDiv         = $('#testDiv')[0];
+    var textarea1       = document.createElement('textarea');
+    var textarea2       = document.createElement('textarea');
+    var textarea3       = document.createElement('textarea');
+    var textarea4       = document.createElement('textarea');
+    var shadowUIElement = document.createElement('textarea');
+
+    shadowUI.addClass(shadowUIElement, 'el');
+
+    testDiv.appendChild(textarea1);
+    testDiv.appendChild(shadowUIElement);
+
+    var elements = document.getElementsByTagName('textarea');
+
+    strictEqual(elements.length, 1);
+    strictEqual(elements[0], textarea1);
+    strictEqual(elements[1], void 0);
+
+    testDiv.appendChild(textarea2);
+    testDiv.appendChild(textarea3);
+    testDiv.appendChild(textarea4);
+
+    strictEqual(elements.length, 4);
```

Comment:
> Most users ...

Who are they? Have you accomplished a research about it? =0 And what about others like me, who has a script on the page that doesn't call `items.length` before getting an item by its index?

Anyway it seems changes are required here. I believe you with @churkin will find the best approach